Bump ioslib, declare xcode 13/ios 15 compat, add ios 15 macro

ewanharris opened the pull request:
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-28467

Related to #12872 which actually makes it work

ewanharris committed:
build(deps): bump to ioslib 1.7.24

Part of TIMOB-28467

ewanharris committed:
feat(ios): declare support for ios 15 and xcode 13

Part of TIMOB-28467

garymathews reviewed (approved):
LGTM!

ewanharris committed:
ci: bump up npm debug level

ewanharris committed:
build: reinstate hasInstallScript onto node-ios-device node

There appears to be an npm bug exposed by this being missing

build commented:
<!--
  1 failure:  Tests have failed...
  0 warning: 
  4 messages
  1 markdown notices
  DangerID: danger-id-Danger;
-->

<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Tests have failed, see below for more information.</td>
    </tr>
  </tbody>
</table>



<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>:fist: The commits in this PR match our conventions! Feel free to Rebase and Merge this PR when ready.</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>:x: 10 tests have failed
There are 10 tests failing and 961 skipped out of 15379 total tests.</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>

  :floppy_disk: <a href="https://jenkins.appcelerator.org/job/titanium-sdk/job/titanium_mobile/job/PR-12873/5/artifact/dist/mobilesdk-10.1.0.v20210616050616-osx.zip">Here's the generated SDK zipfile</a>.
  </td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>:thumbsup: Hey!, You deleted more code than you added. That's awesome!</td>
    </tr>
  </tbody>
</table>

### Tests:

<table><tr><th>Classname</th><th>Name</th><th>Time</th><th>Error</th></tr>
<tr><td>android.emulator.Titanium.UI.ImageView</td><td>.image with redirected URL and autorotate set to true (5.0.2)</td><td>10.758</td><td><pre>Error: Timeout of 10000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (app.js)
</pre></td></tr>
<tr><td>android.emulator.Titanium.UI.Window.methods</td><td>#addSharedElement() (5.0.2)</td><td>10.022</td><td><pre>Error: Timeout of 10000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (app.js)
</pre></td></tr>
<tr><td>android.emulator.Titanium.UI.Window.activity transitions</td><td>TRANSITION_EXPLODE (5.0.2)</td><td>10.022</td><td><pre>Error: Timeout of 10000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (app.js)
</pre></td></tr>
<tr><td>android.emulator.Titanium.UI.Window.activity transitions</td><td>TRANSITION_SLIDE_BOTTOM (5.0.2)</td><td>10.011</td><td><pre>Error: Timeout of 10000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (app.js)
</pre></td></tr>
<tr><td>android.emulator.Titanium.UI.Window.activity transitions</td><td>TRANSITION_SLIDE_TOP (5.0.2)</td><td>10.051</td><td><pre>Error: Timeout of 10000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (app.js)
</pre></td></tr>
<tr><td>android.emulator.Titanium.UI.Window.activity transitions</td><td>TRANSITION_SLIDE_LEFT (5.0.2)</td><td>10.135</td><td><pre>Error: Timeout of 10000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (app.js)
</pre></td></tr>
<tr><td>android.emulator.Titanium.UI.Window.activity transitions</td><td>TRANSITION_SLIDE_RIGHT (5.0.2)</td><td>10.033</td><td><pre>Error: Timeout of 10000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (app.js)
</pre></td></tr>
<tr><td>android.emulator.Titanium.UI.Window.activity transitions</td><td>TRANSITION_FADE_IN/TRANSITION_FADE_OUT (5.0.2)</td><td>10.107</td><td><pre>Error: Timeout of 10000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (app.js)
</pre></td></tr>
<tr><td>ios.macos.Titanium.Blob</td><td>image dimensions should be reported in pixels (10.15.5)</td><td>0.068</td><td><pre>Error: expected 6 to be 11
value@file:///Users/build/jenkins/workspace/ium-sdk_titanium_mobile_PR-12873/tmp/mocha/build/iphone/build/Products/Debug-maccatalyst/mocha.app/Contents/Resources/node_modules/should/cjs/should.js:356:23
postlayout@file:///Users/build/jenkins/workspace/ium-sdk_titanium_mobile_PR-12873/tmp/mocha/build/iphone/build/Products/Debug-maccatalyst/mocha.app/Contents/Resources/ti.blob.test.js:488:33</pre></td></tr>
<tr><td>ios.macos.Titanium.UI.View</td><td>animate width/height from zero (10.15.5)</td><td>10.065</td><td><pre>Error: Timeout of 10000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (app.js)
</pre></td></tr>
</table>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 5a174ce31a93838d79562e5c3faed45235245734
</p>

ewanharris committed:
build(deps): update to ioslib 1.7.26

Fixes TIMOB-28498

hansemannn commented:
Can this be merged? Then the iOS 15 testing could become a bit less stressful each time :-)

ewanharris commented:
@hansemannn, this is kinda blocked for now as I'm currently investigating an issue that's cropped up after the latest rebase where the SDK package script gets stuck in an infinite loop and then crashes with an OOM error

ewanharris committed:
refactor: use aborist to load node_modules tree rather than walking directory

Moves the module-copier to load the node_modules directory via npm's arborist rather than walking
the directory ourselves. This also has the benefit of cutting down on I/O as the data structure
returned from arborist contains information on the dependency type (dev vs prod, optional, peer)
